### ISSUE TITLE
fortios_vpnssl_settings child tables are not cleared on destroy

### ISSUE THREAD

**Issue by poroping** [fortinetdev/terraform-provider-fortios#133]
I believe same issue as #129 The resource is removed without any action on the Fortigate. Leaving stale references which can then not be destroyed as they are referenced by the sslvpn settings.

Apply:
```
resource "fortios_system_interface" "vpn_loopback" {
  provider = fortios.root

  ip          = "157.157.157.157 255.255.255.255" # TODO: DATA SOURCE
  name        = "${var.vdom}-VPN-LOOP"
  type        = "loopback"
  vdom        = var.vdom
  description = "Loopback for VPN termination on ${var.vdom} VDOM"
}

resource "fortios_firewall_address" "sslvpn_user_pool" {
  name       = "${var.vdom}-SSLVPN-USERS-POOL"
  end_ip     = "10.20.50.127" # TODO: DATA SOURCE
  start_ip   = "10.20.50.1"   # TODO: DATA SOURCE
  type       = "iprange"
}

resource "fortios_user_group" "sslvpn_users" {
  group_type = "firewall"
  name       = "SSLVPN-USERS"
}

resource "fortios_vpnsslweb_portal" "blackhole" {
  ipv6_tunnel_mode = "disable"
  name             = "BLACKHOLE"
  tunnel_mode      = "disable"
  web_mode         = "disable"
}

resource "fortios_vpnsslweb_portal" "split_tunnel" {
  forticlient_download = "disable"
  ipv6_tunnel_mode     = "disable"
  name                 = "${var.vdom}-SPLIT-TUNNEL"
  split_tunneling      = "enable"
  tunnel_mode          = "enable"
  web_mode             = "disable"

  ip_pools {
    name = fortios_firewall_address.sslvpn_user_pool.name
  }
}

resource "fortios_vpnssl_settings" "settings" {
  algorithm           = "high"
  auth_timeout        = 28800
  default_portal      = fortios_vpnsslweb_portal.blackhole.name
  https_redirect      = "enable"
  idle_timeout        = 300
  login_attempt_limit = 3
  login_block_time    = 90
  login_timeout       = 90
  port                = 443
  port_precedence     = "enable"

  source_address {
    name = "all"
  }

  source_interface {
    name = fortios_system_interface.vpn_loopback.name
  }

  tunnel_ip_pools {
    name = fortios_firewall_address.sslvpn_user_pool.name
  }

  authentication_rule {
    id     = 1
    portal = fortios_vpnsslweb_portal.split_tunnel.name
    
    groups {
      name = fortios_user_group.sslvpn_users.name
    }
  }
}
```
Fortigate settings:
```
config vpn ssl settings
    set https-redirect enable
    set servercert "Fortinet_Factory"
    set login-attempt-limit 3
    set login-block-time 90
    set login-timeout 90
    set tunnel-ip-pools "TEST-SSLVPN-USERS-POOL"
    set port 443
    set source-interface "TEST-VPN-LOOP"
    set source-address "all"
    set default-portal "BLACKHOLE"
    config authentication-rule
        edit 1
            set groups "SSLVPN-USERS"
            set portal "TEST-SPLIT-TUNNEL"
        next
    end
end
```
Comment out fortios_vpnssl_settings
```
resource "fortios_system_interface" "vpn_loopback" {
  provider = fortios.root

  ip          = "157.157.157.157 255.255.255.255" # TODO: DATA SOURCE
  name        = "${var.vdom}-VPN-LOOP"
  type        = "loopback"
  vdom        = var.vdom
  description = "Loopback for VPN termination on ${var.vdom} VDOM"
}

resource "fortios_firewall_address" "sslvpn_user_pool" {
  name       = "${var.vdom}-SSLVPN-USERS-POOL"
  end_ip     = "10.20.50.127" # TODO: DATA SOURCE
  start_ip   = "10.20.50.1"   # TODO: DATA SOURCE
  type       = "iprange"
}

resource "fortios_user_group" "sslvpn_users" {
  group_type = "firewall"
  name       = "SSLVPN-USERS"
}

resource "fortios_vpnsslweb_portal" "blackhole" {
  ipv6_tunnel_mode = "disable"
  name             = "BLACKHOLE"
  tunnel_mode      = "disable"
  web_mode         = "disable"
}

resource "fortios_vpnsslweb_portal" "split_tunnel" {
  forticlient_download = "disable"
  ipv6_tunnel_mode     = "disable"
  name                 = "${var.vdom}-SPLIT-TUNNEL"
  split_tunneling      = "enable"
  tunnel_mode          = "enable"
  web_mode             = "disable"

  ip_pools {
    name = fortios_firewall_address.sslvpn_user_pool.name
  }
}

# resource "fortios_vpnssl_settings" "settings" {
#   algorithm           = "high"
#   auth_timeout        = 28800
#   default_portal      = fortios_vpnsslweb_portal.blackhole.name
#   https_redirect      = "enable"
#   idle_timeout        = 300
#   login_attempt_limit = 3
#   login_block_time    = 90
#   login_timeout       = 90
#   port                = 443
#   port_precedence     = "enable"

#   source_address {
#     name = "all"
#   }

#   source_interface {
#     name = fortios_system_interface.vpn_loopback.name
#   }

#   tunnel_ip_pools {
#     name = fortios_firewall_address.sslvpn_user_pool.name
#   }

#   authentication_rule {
#     id     = 1
#     portal = fortios_vpnsslweb_portal.split_tunnel.name
    
#     groups {
#       name = fortios_user_group.sslvpn_users.name
#     }
#   }
# }
```
Apply succeeds, resource is 'destroyed':
```
fortios_vpnssl_settings.settings: Destroying... [id=VpnSslSettings]
fortios_vpnssl_settings.settings: Destruction complete after 0s

Apply complete! Resources: 0 added, 0 changed, 1 destroyed.
```
No action is performed on the Fortigate.
Now if we try to destroy we can't due to vpnssl settings referencing the objects we are trying to destroy:
```
Do you really want to destroy all resources?
  Terraform will destroy all your managed infrastructure, as shown above.
  There is no undo. Only 'yes' will be accepted to confirm.

  Enter a value: yes

fortios_system_interface.vpn_loopback: Destroying... [id=TEST-VPN-LOOP]
fortios_user_group.sslvpn_users: Destroying... [id=SSLVPN-USERS]
fortios_vpnsslweb_portal.split_tunnel: Destroying... [id=TEST-SPLIT-TUNNEL]
fortios_vpnsslweb_portal.blackhole: Destroying... [id=BLACKHOLE]

Error: Error deleting VpnSslWebPortal resource: Internal Server Error - Internal error when processing the request (500)



Error: Error deleting SystemInterface resource: Internal Server Error - Internal error when processing the request (500)



Error: Error deleting UserGroup resource: Internal Server Error - Internal error when processing the request (500)



Error: Error deleting VpnSslWebPortal resource: Internal Server Error - Internal error when processing the request (500)
```

**Comment by frankshen01** [comment]
Hi @poroping , thanks for the issue. It will be processed once the previous issues are resolved.

**Comment by frankshen01** [comment]
The issue has been fixed and released, please see the latest version(v1.8.0): https://registry.terraform.io/providers/fortinetdev/fortios/latest, 

**validation:**

```
# cat main.tf
provider "fortios" {
  hostname = "192.168.52.177"
  insecure = "true"
  token    = "rGqsgj9Qmh3dwfQdc8hd3t3G6xG3N5"
}

resource "fortios_vpnssl_settings" "labelname" {
    algorithm                     = "high"
    auth_session_check_source_ip  = "enable"
    auth_timeout                  = 0
    auto_tunnel_static_route      = "enable"
    check_referer                 = "disable"
    default_portal                = "ewiewnca"
    deflate_compression_level     = 6
    deflate_min_data_size         = 300
    dns_server1                   = "0.0.0.0"
    dns_server2                   = "0.0.0.0"
    dtls_hello_timeout            = 10
    dtls_tunnel                   = "enable"
    force_two_factor_auth         = "disable"
    header_x_forwarded_for        = "add"
    http_compression              = "disable"
    http_only_cookie              = "enable"
    http_request_body_timeout     = 30
    http_request_header_timeout   = 20
    https_redirect                = "disable"
    idle_timeout                  = 300
    ipv6_dns_server1              = "::"
    ipv6_dns_server2              = "::"
    ipv6_wins_server1             = "::"
    ipv6_wins_server2             = "::"
    login_attempt_limit           = 2
    login_block_time              = 60
    login_timeout                 = 30
    port                          = 443
    port_precedence               = "enable"
    reqclientcert                 = "disable"
    servercert                    = "self-sign"
    source_address6_negate        = "disable"
    source_address_negate         = "disable"
    ssl_client_renegotiation      = "disable"
    ssl_insert_empty_fragment     = "enable"
    tlsv1_0                       = "disable"
    tlsv1_1                       = "enable"
    tlsv1_2                       = "enable"
    tlsv1_3                       = "enable"
    tunnel_connect_without_reauth = "disable"
    tunnel_user_session_timeout   = 30
    unsafe_legacy_renegotiation   = "disable"
    url_obscuration               = "disable"
    wins_server1                  = "0.0.0.0"
    wins_server2                  = "0.0.0.0"
    x_content_type_options        = "enable"

    authentication_rule {
        auth                   = "any"
        cipher                 = "high"
        client_cert            = "disable"
        id                     = 1
        portal                 = "ewiewnca"
        source_address6_negate = "disable"
        source_address_negate  = "disable"

        source_address {
            name = "2"
        }

        source_address6 {
            name = "1"
        }

        source_interface {
            name = "port2"
        }

        users {
            name = "guest"
        }
    }

    source_address {
        name = "1"
    }

    source_address6 {
        name = "SSLVPN_TUNNEL_IPv6_ADDR1"
    }

    source_interface {
        name = "d1ddd2111cd"
    }
    source_interface {
        name = "port1"
    }
}

# terraform apply

mytestFortiGat # config vpn ssl settings
mytestFortiGate (settings) # show
config vpn ssl settings
    set servercert "self-sign"
    set auth-timeout 0
    set port 443
    set source-interface "d1ddd2111cd" "port1"
    set source-address "1"
    set source-address6 "SSLVPN_TUNNEL_IPv6_ADDR1"
    set default-portal "ewiewnca"
    config authentication-rule
        edit 1
            set source-interface "port2"
            set source-address "2"
            set source-address6 "1"
            set users "guest"
            set portal "ewiewnca"
        next
    end
end

#terraform destroy

mytestFortiGate # config vpn ssl settings

mytestFortiGate (settings) # show
config vpn ssl settings
    set servercert "self-sign"
    set auth-timeout 0
    set port 443
end

```

Thank you!

**Comment by poroping** [comment]
Awesome, tested and works great, thank you!
